### PR TITLE
add: Relationship for IngressClass and other k8s resources

### PR DESCRIPTION
**Pull request by RipulHandoo**
**Notes for Reviewers**

This PR fixes #
1. Adds relationship for Ingress Class and other k8s resources.

https://github.com/meshery/meshery/assets/107461226/41b8e52b-2994-4bc8-8d2a-1c4ecd1bbb0f

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by RipulHandoo** [comment]
_**Note:**_ [Spreadsheet ](https://docs.google.com/spreadsheets/d/1DZHnzxYWOlJ69Oguz4LkRVTFM79kC2tuvdwizOJmeMw/edit#gid=1646054916)updated. ✅

**Comment by pandeyshubham03** [comment]
Looks good to me!

**Comment by leecalcote** [comment]
> Looks good to me!

Thanks for this, @pandeyshubham03 👍

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 1]
```diff
@@ -1077,6 +1077,40 @@
...
+        "from": [
+          {
+            "kind": "IngressClass",
+            "model": "kubernetes",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatorRef": [
+                ["name"]
+              ],
+              "description": "IngressClass establishes an inventory relationship with Ingress. IngressClass is a resource to provide default class for Ingress. When a IngressClass is dropped on Ingress specification of Ingress will be updated and configure the class of the Ingress controller."
```
Comment (leecalcote) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:1092]: Please don't use "dropped". Such phrasing is only applicable to a visual designer interface.

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 2]
```diff
@@ -1174,6 +1174,40 @@
...
+        "to": [
+          {
+            "kind": "Ingress",
+            "model": "kubernetes",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatedRef": [
+                ["settings", "spec", "ingressClassName"]
+              ],
+              "description": "Ingress is a collection of rules that allow inbound connections to reach the services. When a IngressClass is dragged over Ingress specification of Ingress will be updated and configure the class of the Ingress controller"
```
Comment (MUzairS15) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:1202]:
```suggestion
              "description": "Ingress is a collection of rules that allow inbound connections to reach the services. IngressClass configures the class of Ingress resource which is then used by the Ingress Controller."
```

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 3]
```diff
@@ -1174,6 +1174,40 @@
...
+        "from": [
+          {
+            "kind": "IngressClass",
+            "model": "kubernetes",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatorRef": [
+                ["name"]
+              ],
+              "description": "IngressClass establishes an inventory relationship with Ingress. IngressClass is a resource to provide default class for Ingress. When a IngressClass is dragged on Ingress specification of Ingress will be updated and configure the class of the Ingress controller."
```
Comment (MUzairS15) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:1189]:
```suggestion
              "description": "IngressClass is a resource to specify class for Ingress to be used by the IngressController."
```

**Pull request merged**